netkan/netkan/scheduler.py

```diff
@@ -34,14 +34,10 @@ def generate_netkan_message(self, filename):
         }
 
     def sqs_batch_entries(self, batch_size=10):
-        batch = []
-
-        for netkan in self.netkans():
-            batch.append(self.generate_netkan_message(netkan))
-            if len(batch) == batch_size:
-                yield(batch)
-                batch = []
-        yield(batch)
+        netkans = list(self.netkans())
```

Comment:
A path glob pattering is a generator, consuming it like this turns it into a list. If there is nothing in the list, it won't produce an iteration.

```
In [9]: p = Path('/tmp/')  
   ...: for blah in p.glob('*.afds'): 
   ...:     print('blah') 
   ...:                      
                                                                                                                                                                                                                                                                                                
In [10]: type(p.glob('*.afds'))                                                                                                                                                                                                                                                                                              
Out[10]: generator
```
So I'm not sure this is quite the right way to fix it. We're also modifying the list as we're iterating over it, which isn't ideal.

Reply:
> A path glob pattering is a generator, consuming it like this turns it into a list. If there is nothing in the generator, it won't produce an iteration.

Isn't that a good thing? If there are no netkans, we want an empty list here, which will then be skipped by the `while len(netkans) > 0` check.

> So I'm not sure this is quite the right way to fix it. We're also modifying the list as we're iterating over it, which isn't ideal.

What's wrong with modifying the list?

Reply:
> Isn't that a good thing? If there are no netkans, we want an empty list here, which will then be skipped by the `while len(netkans) > 0` check.

Well it collects those all into memory, instead of letting the generator handle it. It likely doesn't matter with < 2000 items.

> What's wrong with modifying the list?

Nevermind, I've absorbed more tea. With a while it is fine, for blows up when you alter the list and it isn't expecting it.

Reply:
OK, latest commit uses a `__list_split` function to break up `self.netkans()` into batches, using what I think is a generator expression so it should be on-demand again. And instead of modifying a local var, we `for` over that generator's values.